core/phases.py

```diff
@@ -25,35 +19,76 @@ def _ready_to_end(self) -> bool:
     def end(self) -> PhaseState:
         if self._ready_to_end():
             self._state = PhaseState.ENDED
+        return self._state
+
+
+class DealPhase(Phase):
+    def __init__(self, game):
+        super().__init__(game)
+
+    def deal_cards(self) -> bool:
+        if self.__can_deal():
+            self.__distribute_cards()
+            self._state = PhaseState.ENDED
```

Comment:
Shouldn't this be PhaseState.READY_TO_END?

Reply:
Yes